### PR TITLE
Added keyboardType & textInputAction props to SearchBar, SearchAnchor & SearchAnchor.bar

### LINKED ISSUE #138483: expose kexboardType and textInputAction parameters of the SearchAnchor TextField
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

For a new Material3 project, I am trying to implement SearchAnchor instead of SearchDelegate. 
However, in contrast to SearchDelegate, the new SearchAnchor appears to be limited in customization options. 
To allow for more customization of the search experience, it would be beneficial to expose parameters similar to the ones of the SearchDelegate. Example parameters are keyboardType or the textInputAction. 

### Proposal

introduce textinputAction and kexboardType as parameters for the SearchBar constructor. Also include them in SearchAnchor() and SearchAnchor.bar() constructors.

### PR DESCRIPTION
**Pull request by piedcipher**
Added `keyboardType` & `textInputAction` props to `SearchBar`, `SearchAnchor` & `SearchAnchor.bar`

Fixes #138483 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/search_anchor.dart
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/search_anchor.dart:1243]: Maybe we can create a {@template ...} ... {@endtemplate} for `TextField.textInputAction` and use {@macro ...} here:)
Flutter style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#dartdoc-templates-and-macros

File: packages/flutter/lib/src/material/search_anchor.dart
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/search_anchor.dart:1246]: Same here and the properties for `SearchAnchor`

**Review by QuncCccccc — COMMENTED** [review]
Thanks so much for the contribution and the great unit test coverage! Just let one comment for the api comments. This looks good to me!

File: packages/flutter/lib/src/material/search_anchor.dart
Comment (piedcipher) [reply packages/flutter/lib/src/material/search_anchor.dart:1243]: Sure...I tried using one template from TextField but it didn't work

Comment (piedcipher) [reply packages/flutter/lib/src/material/search_anchor.dart]: Done 👍✅

File: packages/flutter/lib/src/material/search_anchor.dart
Comment (piedcipher) [reply packages/flutter/lib/src/material/search_anchor.dart]: Done 👍✅

**Review by QuncCccccc — APPROVED** [review]

File: packages/flutter/lib/src/material/search_anchor.dart
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/search_anchor.dart]: Just pushed the change and approved the PR! Once all tests are passed, I will submit it! Thanks again for the improvement:)!

Comment (piedcipher) [reply packages/flutter/lib/src/material/search_anchor.dart]: > Just pushed the change and approved the PR! Once all tests are passed, I will submit it! Thanks again for the improvement:)!

Thank you! 🙌🙌

Comment (QuncCccccc) [reply packages/flutter/lib/src/material/search_anchor.dart]: > keyboardType also has one existing template

Yes, but I think using it directly will cause some failures because the doc uses `{maxlines}` which is a property of `TextField` but not a property of `SearchBar`

**Comment by piedcipher** [comment]
I think I'll need one more approval. ✅ ✔️

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**
